cli/tools/task.rs

```diff
@@ -105,24 +106,85 @@ pub async fn execute_script(
         .unwrap()
         .to_owned(),
     };
-    let script = get_script_with_args(script, cli_options);
-    output_task(task_name, &script);
-    let seq_list = deno_task_shell::parser::parse(&script)
-      .with_context(|| format!("Error parsing script '{task_name}'."))?;
-    let npx_commands = resolve_npm_commands(npm_resolver, node_resolver)?;
-    let env_vars = collect_env_vars();
-    let local = LocalSet::new();
-    let future =
-      deno_task_shell::execute(seq_list, env_vars, &cwd, npx_commands);
-    let exit_code = local.run_until(future).await;
-    Ok(exit_code)
+
+    let mut pre_task_name: String = "pre".to_owned();
+    pre_task_name.push_str(task_name);
+    if let Some(pre_script) = package_json_scripts.get(&pre_task_name) {
```

Comment:
If you run `npm run pretest` does it try to run `prepretest`?

Reply:
Yup npm will try to run `prepretest` if it's available.

```json
{
  "scripts": {
    "prepretest": "echo 'prepretest'",
    "pretest": "echo 'pretest'",
    "posttest": "echo 'posttest'",
    "test": "echo 'test'"
  }
}
```

Output:

```sh
$ npm run pretest

> prepretest
> echo 'prepretest'

prepretest

> pretest
> echo 'pretest'

pretest
```